### PR TITLE
chore: Code split for saving and updating action

### LINKED ISSUE #27563: [Task]: Split code for saving and updating action
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description
There's a dependency on pageId and newPage in the implementation of creating or updating action. Creation flow of module will also create a document in the newAction collection. The expectation here is to use the underlying implementation. We need to introduce code split here to make sure module creation flow can leverage this implementation.

### SubTasks

- Extract validation related to `pageId` and move it to a new method so that EE code can override this function w.r.t. module's requirement

### PR DESCRIPTION
**Pull request by subrata71**
## Description

There's a dependency on pageId and newPage in the implementation of creating or updating action. Creation flow of module will also create a document in the newAction collection. The expectation here is to use the underlying implementation. We need to introduce code split here to make sure module creation flow can leverage this implementation.

#### PR fixes following issue(s)
Fixes #27563 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString
+public class ActionCE_DTO implements Identifiable, Executable {
```
Comment (trishaanand) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java:37]: Should ActionDTO instead implement Identificable, Executable

**Review by trishaanand — COMMENTED** [review]
Minor comment added. LGTM.

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java (hunk 1 above)
Comment (subrata71) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java:37]: CE DTO implements interfaces that exist in the CE code. EE DTO will implement EE specific interface. For example, ModuleConsumable interface exists in the EE code, hence ActionDTO will implement this interface. @trishaanand

Comment (trishaanand) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java:37]: Considering the fact that ActionDTO is what gets used across the services (and not ActionCE_DTO), I was thinking that ActionDTO implements Identifiable, Executable on CE and Identifiable, Executable, ModuleConsumable on EE. Does that make sense?

Comment (subrata71) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java:37]: It absolutely makes sense but if `CE_DTO` implements the CE-related interfaces and EE_DTO implements EE specific interfaces then it leads to less merge conflict. A recent example would be - Nidhi introduced `Executable` interface implemented by `ActionDTO` in the CE and I was working on in the EE repo at the same time and introduced `ModuleConsumable` interface. 

This resulted in a merge conflict. Please let me know if you think still we should make the change as you suggested. I see points in both the approaches. 

cc: @nidhi-nair

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCE.java [hunk 2]
```diff
@@ -141,4 +141,8 @@ Mono<ImportedActionAndCollectionMapsDTO> updateActionsWithImportedCollectionIds(
     Mono<UpdateResult> publishActions(String applicationId, AclPermission permission);
 
     Flux<PluginTypeAndCountDTO> countActionsByPluginType(String applicationId);
+
+    void setGitSyncIdInNewAction(NewAction newAction);
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCE.java:145]: Does this need to be part of the interface? Is it not a private function? Same for the next method.

**Review by nidhi-nair — COMMENTED** [review]
Tiny comment on reducing visibility of methods.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCE.java (hunk 2 above)
Comment (subrata71) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCE.java:145]: it needs to be overridden in the EE as the generation of gitSyncId is currently based on this logic `newAction.getApplicationId() + "_" + new ObjectId()`. And hence it's part of the interface.

Comment (trishaanand) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCE.java:145]: You dont have to include a function that needs to be overridden in EE. You can use protected function and override the same in the EE Impl file. @subrata71

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**